server/admin_management/api/app/services/permission_audit_service.py

```diff
@@ -60,6 +61,31 @@ def store_delegated_admin_permissions_granted_audit_history(
         LOGGER.debug(f"Adding audit record for ({change_type}): {audit_record}")
         self.repo.save(audit_record)
 
+    def store_application_admin_permission_granted_audit_history(
+        self,
+        requester: Requester,
+        change_target_user: FamUser,
+        fam_application_admin_user: FamApplicationAdmin
+    ):
+        change_type = PrivilegeChangeTypeEnum.GRANT
+        privilege_details = PrivilegeDetailsSchema(
+            permission_type=PrivilegeDetailsPermissionTypeEnum.APPLICATION_ADMIN,
+        )
+        audit_record = PermissionAuditHistoryCreateSchema(
+            application_id=fam_application_admin_user.application.application_id,
```

Comment:
This line to obtain the application id looks weird.

Reply:
Why does the admin user's application matter?

Reply:
When application admin is granted we need to store the audit record for this.
To save the audit record, one attribute to be saved is the application_id, and this can be obtained from the currently granted `fam_application_admin_user` (this is mapped to the `fam_application_admin` table entity) from it's `application` relationship to grab the id.

Reply:
Ah, I think I was confused between the admin user performing the action and the admin access being granted, which is really a permission being granted and not a user, so I was confused by it being called fam_application_admin_user.

Reply:
I see, the variable name confuses your thought.
I like to refactor the variable name (not only on this file context, some other files too related to this name), what name would you give or prefer so that you don't get confused?
The table entity class is called `FamApplicationAdmin` and the variable is associated with that type, perhaps removing `_user` at the end (`fam_application_admin`), will it be less confusing to you (or maybe, `fam_application_admin_grant`?

Reply:
Yes, removing _user suffix would likely be clearer.

Reply:
Variable is renamed to `fam_application_admin` without _user suffix. Thanks Basil!